client/lib/feed-stream-store/index.js

```diff
@@ -152,8 +153,27 @@ function getStoreForFeatured( storeId ) {
 
 function getStoreForRecommendedPosts( storeId ) {
 	var fetcher = function( query, callback ) {
-			wpcomUndoc.readRecommendedPosts( query, callback );
-		};
+			function trainTracksProxy( err, response ) {
+				var eventName= 'calypso_traintracks_render',
+					eventProperties = {};
+				if ( response && response.posts ) {
```

Comment:
this can use `lodash/forEach` and avoid all the checks. Just `forEach( response && response.posts, post => { ... } )`